### ISSUE TITLE
crash with IllegalArgumentException: requirement failed

### ISSUE THREAD

**Issue by helloqirun** [lampepfl/dotty#1663]
$ cat abc.scala
```
import scala.annotation
class ann()
object Api {
  @ann({ def foo })
  def foo()
}
```

$ dotc abc.scala
```
-- [E018] Syntax Error: abc.scala --------------------------------------------------------------------------------------
5 |  @ann({ def foo })
  |                ^
  |                missing return type

longer explanation available when compiling with `-explain`
-- [E018] Syntax Error: abc.scala --------------------------------------------------------------------------------------
6 |  def foo()
  |           ^
  |           missing return type

longer explanation available when compiling with `-explain`

exception occurred while typechecking abc.scala

exception occurred while compiling abc.scala
Exception in thread "main" java.lang.IllegalArgumentException: requirement failed
        at scala.Predef$.require(Predef.scala:207)
        at dotty.tools.dotc.core.SymDenotations$ClassDenotation.enterNoReplace(SymDenotations.scala:1537)
        at dotty.tools.dotc.core.SymDenotations$ClassDenotation.enter(SymDenotations.scala:1527)
        at dotty.tools.dotc.typer.NamerContextOps$class.enter(Namer.scala:34)
        at dotty.tools.dotc.core.Contexts$Context.enter(Contexts.scala:57)
        at dotty.tools.dotc.typer.Namer.enterSymbol(Namer.scala:334)
        at dotty.tools.dotc.typer.Namer.dotty$tools$dotc$typer$Namer$$recur$1(Namer.scala:417)
        at dotty.tools.dotc.typer.Namer.indexExpanded(Namer.scala:427)
        at dotty.tools.dotc.typer.Namer$$anonfun$6.apply(Namer.scala:522)
        at dotty.tools.dotc.typer.Namer$$anonfun$6.apply(Namer.scala:522)
        at scala.collection.LinearSeqOptimized$class.foldLeft(LinearSeqOptimized.scala:124)
        at scala.collection.immutable.List.foldLeft(List.scala:84)
<snippet>
```

**Comment by liufengyun** [comment]
Thanks @helloqirun . This seems to duplicate with #1647. I leave it open for now, so that we can check if the fix to #1647 also fix this one.

**Comment by helloqirun** [comment]
Ah, I see. Sorry for that, I forgot I have reported that a few days ago. Will check it more careful next time.
